2016/proposals/verma-nikhil.md

```diff
@@ -0,0 +1,232 @@
+
+## Proposed Workflow
+
+The whole system to manage instructor training is supposed to have two interfaces:
+
+1. The Trainee Admin - An administrative interface - which is used by admins to manage trainees at each stage, and
+
+2. The Trainee Dashboard - the one used by trainees to keep track of their progress.
+
+The process starts with applicants applying for instructor training programmes
+
+### 1. The Application Form
+
+The application process for becoming an instructor starts here. People aspiring to become instructors shall start by filling up this application form.
+
+Trainees apply in one of the following ways:
+
+  1. **Apply individually**:
+
+  This would be a normal application, requesting the relevant details from the applicant.
+
+  **Helping individuals form groups**
+
+   * Apart from the regular details, the applicant would also be asked if they are willing to pair up.
+   * If yes, they would be presented with a list of individuals or groups willing to accept new members. The applicant can then request to pair up with the individual/group of their choice.
+
+  **Determine alredy associated applicants**
+
+  The applicant will be presented with a list of partner organizations which will be used to determine whether they are already associated with a partner organization
+
+  2. **Apply in a group**
+
+  Applicants can also apply in groups, making it easier to manage a bunch of people together. This would have the following fields:
+
+   * Individual details of every member, as in [1] above
+   * Group Leader
+   * Whether or not they are willing to expand the group
+
+
+### 2. Managing Trainees - The Trainee Admin
+
+  The Trainee Admin will be used to manage trainees at each stage of training. Access will be based on permissions (to be discussed during GSoC), so that the right people get to change and monitor the trainee status at each stage. For now, an *admin* is the relevant person handling a particular stage for an applicant. The admin workflow is outlined here:
+
+  * All the applications will be visible to *admins* on a dashboard.
+  * The *admins* start by reviewing applications for instructor training that were filled in the form above.
+  * They can then shortlist applicants and match them to courses. Multiple applications can be updated at once for ease.
+  * Once this is done, the applicants are sent an invitation to attend the selected course and also to join the Trainee Dashboard.
+
+  >Trainees log in, and attend the course
+
+  * The *admin* updates course completion status
+
+  >Trainees submit homework link
+
+  * The *admin* reviews the homework, and provide necessary feedback. Once they are satisfied, they update homework completion status.
+
+  >Trainee provides preferred time for discussion session (optional)
```

Comment:
If the lesson maintainer approves changes instead of the trainer, it's just a matter of changing permissions regarding who approves a homework. We'd just have to modify the `@permission_required` decorator. Most likely, this is going to be a one line change.

Coming to user comments, both, GitHub and Google Docs support comments. It would be easier to use those natively.

If the participation is recorded after a discussion session, the discussion leader can mark the attendance on their admin interface or (if possible) have some assistant do that for them.